langchain/vectorstores/pinecone.py

```diff
@@ -95,6 +96,59 @@ def add_texts(
         self._index.upsert(vectors=docs, namespace=namespace, batch_size=batch_size)
         return ids
 
+    def search_documents_by_metadata(
```

Comment:
> I realize this might seem a bit abstract without a practical demonstration, so if you think it would be helpful, I'd be more than happy to whip up a Jupyter notebook.

Hmm, yes please add notebook to clarify.

Pinecone already [accepts metadata filter ](https://github.com/hwchase17/langchain/blob/bd0c6381f5a6ead453f4c5576f45988a53c28aa5/langchain/vectorstores/pinecone.py#L132)as a dict.

Also, we have [self-query retriever](https://python.langchain.com/docs/modules/data_connection/retrievers/how_to/self_query/) that infers metadata from a query. 

Do either

Reply:
Like @levalencia mentions below, IIUC this is achieved w/ self query retriever:
https://python.langchain.com/docs/modules/data_connection/retrievers/how_to/self_query/